src/features/Table/index.tsx

```diff
@@ -3,20 +3,44 @@
 // import SearchBar from '../SearchBar';
 import Row from './Row';
 
-const Table = (): JSX.Element => {
+export interface TableEmployee {
+  id: string | number
+  name: string
+  email: string
+  role: string
+};
+
+interface IProps {
+  employees: TableEmployee[]
+  onSelect?: (employee: TableEmployee) => void
+}
+
+const Table = ({
+  employees,
+  onSelect,
+}: IProps): JSX.Element => {
+  const isClickable = !(onSelect == null);
+
+  const handleOnClick = (employee: TableEmployee): void => {
+    if (onSelect != null) {
+      onSelect(employee);
+    }
+  };
+
+  const employeeRows = employees.map((employee) => {
+    return (
+      <Row key={`employee-${employee.id}`} data={[employee.name, employee.email, employee.role]} onClick={() => handleOnClick(employee)} isClickable={isClickable} />
+    );
+  });
```

Comment:
It was a static hard coded and I had only to adjust it to have it more dynamic in order to create a dynamic checklist. I 

I could not identify how this kind of static code was is usable for another components. It seemed that it just a temporary employees data to show an example of how the dashboard should look like .  

This is how it's in main now : 

import Row from './Row';

const Table = (): JSX.Element => {
  return (
    <div className='rounded-sm overflow-hidden border border-gray-100 bg-white'>
      <table className="table-auto border-collapse w-full">
        <thead className="bg-gray-100">
          <Row data={['Employee name', 'Email', 'Role']} head/>
        </thead>
        <tbody>
          <Row data={['John Doe', 'j.doe@email.com', 'Admin']}/>
          <Row data={['John Doe', 'j.doe@email.com', 'Admin']}/>
          <Row data={['John Doe', 'j.doe@email.com', 'Admin']}/>
          <Row data={['John Doe', 'j.doe@email.com', 'Admin']}/>
          <Row data={['John Doe', 'j.doe@email.com', 'Admin']}/>
          <Row data={['John Doe', 'j.doe@email.com', 'Admin']}/>
        </tbody>
      </table>
    </div>
  );
};

export default Table;

Reply:
Like I mention to you at the when you started this task, the code was hard coded in the main branch. It does not remove the fact that this table component is and will be used by other page in the application. The Activity page will be using this component. Right now it has hard coded the information inside a copy of the table because the link with the array of rows had not been made yet. Once the connection is done, the code will be refactor to remove the duplicates and the component will be used by the ActivityPage.

Reply:
In this case , the best is to wait for this connection to be implemented and pushed to production , and then merge the delete employee feature .

what would you say ?

Reply:
That is what you are doing. I would say you are 90% there but if can't extract the specific of the employee. I will do it next week by refactoring the table.